src/transformation/visitors/loops/do-while.ts

```diff
@@ -1,23 +1,53 @@
 import * as ts from "typescript";
 import * as lua from "../../../LuaAST";
 import { FunctionVisitor } from "../../context";
-import { transformLoopBody } from "./utils";
+import { transformInPrecedingStatementScope } from "../../utils/preceding-statements";
+import { invertCondition, transformLoopBody } from "./utils";
 
-export const transformWhileStatement: FunctionVisitor<ts.WhileStatement> = (statement, context) =>
-    lua.createWhileStatement(
-        lua.createBlock(transformLoopBody(context, statement)),
-        context.transformExpression(statement.expression),
-        statement
+export const transformWhileStatement: FunctionVisitor<ts.WhileStatement> = (statement, context) => {
+    const body = transformLoopBody(context, statement);
+
+    let [precedingStatements, condition] = transformInPrecedingStatementScope(context, () =>
+        context.transformExpression(statement.expression)
     );
 
+    // Change from 'while condition' to 'while true - [preceding statements] if not condition then break'
+    // This is so that preceding statements are evaluated every iteration.
```

Comment:
Could you specify here this is about the preceding statements of the condition? Could you maybe also add this 'reason' comment to the other instances where you do this? (line 39 of this file, 31 in for.ts)